lib/Service/ProvisioningService.php

```diff
@@ -58,6 +58,54 @@ public function hasOidcUserProvisitioned(string $userId): bool {
 		return false;
 	}
 
+	/**
+	 * Resolves a claim path like "custom.nickname" or multiple alternatives separated by "|".
+	 * Returns the first found string value, or null if none could be resolved.
+	 */
+	public function getClaimValue(object|array $tokenPayload, string $claimPath, int $providerId): ?string {
+		if ($claimPath === '') {
+			return null;
+		}
+
+		// Check config if dot-notation resolution is enabled
+		$resolveDot = false;
+		$resolveDot = $this->providerService->getSetting($providerId, ProviderService::SETTING_RESOLVE_NESTED_AND_FALLBACK_CLAIMS_MAPPING, false);
```

Comment:
```suggestion
		$resolveDot = $this->providerService->getSetting($providerId, ProviderService::SETTING_RESOLVE_NESTED_AND_FALLBACK_CLAIMS_MAPPING, '0') === '1';
```

Reply:
I took the freedom to push in your branch. I hope that's fine. So the `getSetting` call is fixed.